File: DailyChallenge/2020/March/27.cpp
// https://leetcode.com/explore/challenge/card/march-leetcoding-challenge-2021/591/week-4-march-22nd-march-28th/3686/

class Solution {
public:
    int countSubstrings(string s) 
    {
        int count=0;
        int n=s.size();
        
        for (int i=0; i<s.size(); i++)
        {
            int len=0;
            while (i-len>=0 && i+len<n && s[i-len]==s[i+len])
                len++;
            count+=len;
            
            len=0;
            while (i-len>=0 && i+1+len<n && s[i-len]==s[i+1+len])
                len++;
            count+=len;
        }
        return count;

    }
};

File: DailyChallenge/2020/April/27.java
// https://leetcode.com/explore/challenge/card/april-leetcoding-challenge-2021/596/week-4-april-22nd-april-28th/3722/

class Solution {
    public boolean isPowerOfThree(int n) {
        if (n < 1) {
            return false;
        }

        while (n % 3 == 0) {
            n /= 3;
        }
        return n == 1 ? true : false;
    }
}
